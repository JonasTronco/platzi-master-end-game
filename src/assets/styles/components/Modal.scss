@import '../main.scss';

.modal {
  @include colCenCen;
  z-index: 9999999;
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background-color: $modal_color;
}

.modal__close {
  position: absolute;
  right: 30px;
  top: 30px;
  height: 30px;
  width: 30px;
  fill: $white_color;
  cursor: pointer;
}

.modal__close:hover {
  fill: $negative_color;
}

.modal__close:active {
  transform: scale(0.95);
}
