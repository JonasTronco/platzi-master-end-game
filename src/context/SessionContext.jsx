import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
