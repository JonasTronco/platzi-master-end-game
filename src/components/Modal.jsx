import PropTypes from 'prop-types';
import '../assets/styles/components/Modal.scss';

function Modal({ isOpen, children }) {
  if (!isOpen) {
    return null;
  }

  return (
    <div className="modal">
      <div className="modal__container">{children}</div>
    </div>
  );
}

Modal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
};

export default Modal;
