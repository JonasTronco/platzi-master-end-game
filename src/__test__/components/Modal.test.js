import Modal from '../../components/Modal';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import { SessionProvider } from '../../context/SessionContext';
describe('Modal Component', () => {
  test('Should render Modal component', () => {
    const component = create(
      <SessionProvider>
        <Modal isOpen={true}>
          <div></div>
        </Modal>
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('<Modal />', () => {
    const component = mount(
      <SessionProvider>
        <Modal isOpen={true}>
          <div></div>
        </Modal>
      </SessionProvider>
    );
    expect(component.length).toEqual(1);
  });
});
