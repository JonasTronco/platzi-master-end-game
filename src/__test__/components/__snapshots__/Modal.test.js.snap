// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component Should render Modal component 1`] = `null`;
